.github/workflows/terraform-ci.yml

```
@@ -121,7 +125,7 @@ jobs:
           --junitfile "$TEST_RESULTS/gotestsum-report.xml" \
           --format standard-verbose -- \
           ./... -p 1 -timeout 90m -v -failfast
-    - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+    - uses: actions/upload-artifact@v4
```

We should pin the GitHub Action for better security.

```suggestion
    - uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 #v4.6.2
```